src/ui/pages/BridgeForm/getAvailablePositions.ts

```diff
@@ -0,0 +1,34 @@
+import { sortPositionsByValue } from '@zeriontech/transactions';
+import type { AddressPosition } from 'defi-sdk';
+import type { Chain } from 'src/modules/networks/Chain';
+
+export function getAvailablePositions({
+  positions,
+  supportedTokens,
+  chain,
+}: {
+  positions?: AddressPosition[] | null;
+  supportedTokens?: string[] | null;
+  chain?: Chain | null;
+}) {
+  if (!positions || !supportedTokens || !chain) {
+    return null;
+  }
+
+  const supportedTokensSet = new Set(supportedTokens);
+  const availablePositions = positions.filter(
+    (position) =>
+      position.type === 'asset' &&
+      position.chain === chain.toString() &&
+      supportedTokensSet.has(position.asset.asset_code)
+  );
+
+  const sorted = sortPositionsByValue(availablePositions);
+  const map: Record<string, AddressPosition> = {};
+
+  for (const position of sorted) {
+    map[position.asset.asset_code] = position;
+  }
+
+  return { sorted, map };
```

Comment:
```
map: Object.fromEntries(sorted.map(position => ([position.asset.asset_code, position])));
```

Reply:
However, not important :)

Reply:
this code block used the for loop before I extracted the function, so I left this unchanged,
you think its worth changing to` Object.fromEntries + .map`? the loop should be more efficient afaik